libs/common/src/platform/services/state.service.ts

```diff
@@ -2450,6 +2450,23 @@ export class StateService<
     )?.data?.providers;
   }
 
+  async getPreviousUrl(options?: StorageOptions): Promise<string> {
+    return (
+      await this.getGlobals(this.reconcileOptions(options, await this.defaultOnDiskLocalOptions()))
```

Comment:
I'm not sure what the implications are int eh extension but in the web client there is an issue using the `defaultOnDiskLocalOptions` as it uses Local Storage instead of Session Storage. If a user has two tabs open the local storage is shared between them causing usability issues with navigation. 

My concern is by saving `previousUrl` into Local Storage it may impact the web client. I would suggest using the `defaultOnDiskOptions` instead, since it uses session storage instead of local storage. Which will restrict the scope of the `GlobalState.previousUrl` to the tab you are in.

```suggestion
      this.reconcileOptions(options, await this.defaultOnDiskOptions())
```